compiler/ballerina-tools-api/src/main/java/io/ballerina/tools/diagnostics/DiagnosticInfo.java

```diff
@@ -64,4 +74,8 @@ public DiagnosticSeverity severity() {
     public Collection<DiagnosticRelatedInformation> relatedInformation() {
         return relatedInformation;
     }
+
+    public Object[] getArgs() {
```

Comment:
Args being `Object` api-consumer need to call `instanceOf`. Can't we have a specific thing for symbol-info where the type is `Symbol`

Reply:
In BlangDiagnosticLog usages (such as `error()` method), sometimes it takes `BTypes` and sometimes `Strings`. That's why I have allowed for `Object`. Any alternative way to do this ?

Reply:
Yes, IMO at this phase, we cannot migrate all diagnostics into args based templates. Even we do, there's no point in retrieving `string` args since it does not contain any symbol meta-data. What diagnostics messages missing are the symbol-info. Thus, it should be symbol-info which are optional and may provided by some diagnostics such as incompatible types.